rclrs/src/parameter/value.rs

```diff
@@ -319,6 +324,107 @@ impl ParameterVariant for ParameterValue {
     }
 }
 
+impl From<ParameterValue> for RmwParameterValue {
+    fn from(value: ParameterValue) -> Self {
+        match value {
+            ParameterValue::Bool(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_BOOL,
+                bool_value: v,
+                ..Default::default()
+            },
+            ParameterValue::Integer(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_INTEGER,
+                integer_value: v,
+                ..Default::default()
+            },
+            ParameterValue::Double(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_DOUBLE,
+                double_value: v,
+                ..Default::default()
+            },
+            ParameterValue::String(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_STRING,
+                string_value: v.into(),
+                ..Default::default()
+            },
+            ParameterValue::ByteArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_BYTE_ARRAY,
+                byte_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::BoolArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_BOOL_ARRAY,
+                bool_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::IntegerArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_INTEGER_ARRAY,
+                integer_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::DoubleArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_DOUBLE_ARRAY,
+                double_array_value: (*v).into(),
+                ..Default::default()
+            },
+            ParameterValue::StringArray(v) => RmwParameterValue {
+                type_: ParameterType::PARAMETER_STRING_ARRAY,
+                string_array_value: v
+                    .iter()
+                    .map(|v| v.clone().into())
+                    .collect::<Vec<_>>()
```

Comment:
Since `Sequence` implements `FromIterator`, I think you can change these next two lines into simply
```
.collect();
```
That will avoid an unnecessary heap allocation to produce a `Vec` followed by another heap allocation to move everything over into the sequence.

Reply:
Done https://github.com/ros2-rust/ros2_rust/pull/342/commits/a7d1befbe078418fca4852b200e3e1b6c4a62f9f